packages/core/admin/admin/src/pages/SettingsPage/components/SettingsNav/index.js

```diff
@@ -8,9 +9,16 @@ import {
   SubNavSections,
 } from '@strapi/design-system/v2';
 import { useTracking } from '@strapi/helper-plugin';
+import { Lock } from '@strapi/icons';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 import { NavLink, useLocation } from 'react-router-dom';
+import styled from 'styled-components';
+
+const CustomIcon = styled(Icon)`
+  right: 15px;
+  position: absolute;
```

Comment:
can you not use a flex container? this feels like a CSS hack

Reply:
How do you want to use Flex here? It is either putting the icon below or making the menu link background shorter using a gap 🤔

Reply:
Okay i've had a look at this, our component is really badly built.

Can you amend with this comment:

```js
/**
 * TODO: refactor the SubNav entirely, we shouldn't have
 * to do this hack to work a lock at the end. It's a bit hacky.
 */
```

Not a comment on your work by the way – just need to remind myself in the future :)

Reply:
Sure :)